File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    [(3, "Pling"); (5, "Plang"); (7, "Plong")]
    |> List.map (fun (factor, sound) -> if number % factor = 0 then sound else "")
    |> String.concat ""
    |> fun result -> if result = "" then string number else result

File: ocaml/raindrops/raindrops.ml

let make_sound noise div num =
  if num mod div = 0
  then noise
  else String.empty

let pling =
  make_sound "Pling" 3

let plang =
  make_sound "Plang" 5

let plong =
  make_sound "Plong" 7

let raindrop num =
    let result =
      [pling num; plang num; plong num]
      |> String.concat String.empty in

    if String.length result = 0
    then string_of_int num
    else result
